src/main/java/de/tum/in/www1/artemis/repository/ProgrammingExerciseStudentParticipationRepository.java

```diff
@@ -27,7 +27,8 @@
     Optional<ProgrammingExerciseStudentParticipation> findByIdWithLatestResultAndFeedbacksAndRelatedSubmissions(@Param("participationId") Long participationId,
             @Param("dateTime") ZonedDateTime dateTime);
 
-    @Query("select p from ProgrammingExerciseStudentParticipation p left join fetch p.results pr left join fetch pr.feedbacks left join fetch pr.submission left join fetch pr.assessor where p.id = :participationId and (pr.assessmentType = 'MANUAL' or pr.assessmentType = 'SEMI_AUTOMATIC')")
+    @Query("select p from ProgrammingExerciseStudentParticipation p left join fetch p.results pr left join fetch pr.feedbacks left join fetch pr.submission left join fetch pr.assessor"
+            + " where p.id = :participationId and pr.id = (select max(prr.id) from p.results prr where prr.assessmentType = 'MANUAL' or prr.assessmentType = 'SEMI_AUTOMATIC')")
```

Comment:
Could we use better names instead of `pr` and `prr`? At least the last one does express what it is in my eyes, or is this to indicate that we have multiple results at this point? Would `prs` be better or worse?

Reply:
This is also used at other places